From c9346bfbb93759b54a9324067e232546805c1cd1 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 15 Nov 2021 15:12:54 +0800
Subject: [PATCH 57/93] gl-renderer: dmabuf: Support NV12(AFBC) and
 NV12_10(AFBC)

Support importing DRM_FORMAT_YUV420_(8|10)BIT dmabuf.

Tested on RK356x with:
gst-play-1.0 video.mp4 --videosink=waylandsink

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/renderer-gl/gl-renderer.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libweston/renderer-gl/gl-renderer.c b/libweston/renderer-gl/gl-renderer.c
index 2782e66..259b24d 100644
--- a/libweston/renderer-gl/gl-renderer.c
+++ b/libweston/renderer-gl/gl-renderer.c
@@ -2647,6 +2647,8 @@ choose_texture_target(struct gl_renderer *gr,
 	case DRM_FORMAT_VYUY:
 	case DRM_FORMAT_AYUV:
 	case DRM_FORMAT_XYUV8888:
+	case DRM_FORMAT_YUV420_8BIT:
+	case DRM_FORMAT_YUV420_10BIT:
 		return GL_TEXTURE_EXTERNAL_OES;
 	default:
 		return GL_TEXTURE_2D;
-- 
2.20.1

